services/order.py

```diff
@@ -0,0 +1,29 @@
+from django.contrib.auth import get_user_model
+
+from db.models import Order, Ticket, MovieSession
+from django.db import transaction
+
+
+@transaction.atomic
+def create_order(
+    tickets: list[dict], username: str, date: str | None = None
```

Comment:
The type hint `list[dict]` is used here, which is only supported in Python 3.9 and later. Ensure that your environment supports this syntax or use `List[Dict]` from `typing` for compatibility with earlier versions.

Reply:
agree